tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/SearchAPI.java

```diff
@@ -0,0 +1,155 @@
+import org.eclipse.microprofile.openapi.annotations.enums.SchemaType;
+import org.eclipse.microprofile.openapi.annotations.media.Content;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+import org.eclipse.microprofile.openapi.annotations.parameters.Parameter;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponse;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponses;
+import org.eclipse.microprofile.openapi.annotations.tags.Tag;
+import org.unicode.cldr.util.CLDRLocale;
+import org.unicode.cldr.web.CookieSession;
+import org.unicode.cldr.web.SearchManager;
+import org.unicode.cldr.web.SearchManager.SearchRequest;
+import org.unicode.cldr.web.SearchManager.SearchResponse;
+
+/**
+ * API for querying SurveyTool
+ * @see {@link SearchManager}
+ */
+@Path("/search")
+@Tag(name = "searching", description = "APIs for Searching")
+public class SearchAPI {
+
+    protected static final class SearchAPIHelper {
+        final SearchManager searchManager = SearchManager.forFactory(CookieSession.sm.getSTFactory());
+        public final static SearchAPIHelper INSTANCE = new SearchAPIHelper();
+
+        public static final SearchManager getSearchManager() {
+            return INSTANCE.searchManager;
+        }
+    };
+
+    @Path("/value")
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Operation(
+        summary = "Begin a new search",
+        description = "Searches for paths containing the given string value")
+    @APIResponses(
+        value = {
+            @APIResponse(
+                responseCode = "200",
+                description = "Perform search",
+                content = @Content(mediaType = "application/json",
+                    schema = @Schema(implementation = SearchResponse.class))),
+            @APIResponse(
+                responseCode = "401",
+                description = "Unauthorized - need to be logged in as a valid user"),
+            @APIResponse(
+                responseCode = "404",
+                description = "Locale not found"),
+        })
+    public Response newSearch(
+        @Parameter(
+            required = false, example = "jgo", schema = @Schema(type = SchemaType.STRING)) @QueryParam("loc") String loc,
+        @HeaderParam(Auth.SESSION_HEADER) String session,
+        SearchRequest request) {
+        final CookieSession mySession = Auth.getSession(session);
+        // User must be logged in to use query function
+        if (mySession == null || mySession.user == null) {
+            return Auth.noSessionResponse();
+        }
+
+        final SearchManager searchManager = SearchAPIHelper.getSearchManager();
+
+        if (!CookieSession.sm.isValidLocale(CLDRLocale.getInstance(loc))) {
+            return Response.status(Status.NOT_FOUND).build();
+        }
+        SearchResponse search = searchManager.newSearch(request, loc);
+        // TODO: pre-validate response?
```

Comment:
what does "pre-validate response" mean?

Reply:
It means to check the response to see if it had a failure. No failure modes yet, so i'll remove the comment.